### ISSUE TITLE
[BUG] Inline styles from CMS not in SSR HTML causing a big layout shift

### ISSUE THREAD

**Issue by niklaswolf** [vuestorefront/shopware-pwa#1303]
### Describe the bug

I described the issue in the slack channel here https://vuestorefront.slack.com/archives/CT7AQR695/p1610447010424500
but I'll try to add a description here:

We have problems with CMS content that contains inline style-attributes: the style attributes do not get into the SSR html, only after hydration on the client side, which then results in a huge layout shift.
You can reproduce this with the `CmsElementText` component as follows:
```js
render(h, context) {
    const config = {...}
    // const rawHtml = context.props.content?.data?.content
    // Version 1: Works, style attribute is in SSR html
    const element = h("h1",{style: "text-align: center; color: #fff", domProps: {
        innerHTML: 'HElloooooo'
      }});
    return element;
    // Version 2: Doesn't work, style attribute not in SSR html
    const rendered = renderHtml("<h1 style='text-align: center; color: #fff'>HElloooooo</h1>", config, h, context)
    return rendered;
  },
  ```
Version 2 is like it’s done right now, the html code from the cms gets passed to `renderHtml` .
Version 1 is a test to render the same element directly.

So this looks like a bug in the `renderHtml` function, but I think this is also something someone from your team wrote, am I right?

### Additional context
What I saw is that the style attribute apparently exists on the VNode returned by the `renderHtml` function, but it somehow ends up NOT being rendered in the final html returned to the client.
This seems like problem that has to be fixed soon in my view. Especially when the shop-owner styles many things using the wysiwyg editor, this will be more visible on the frontend side as it results in big repaints by the browser etc.

**Comment by mkucmus** [comment]
After small investigation I can say that applying the css inline styles (comming right with v-html which is raw) are being done on mounted event, not before. Looks like a bug in the vue itself. It's probably caused by that mechanism when vue make a copy of current dom into virtual dom and then the events are fired... (styles from inline and component itself are being merged).


quick soulution would be like wrap into transition (for ui feeling only) and conditionally render the component after the mounted event is done, we then have hydrated cms component with all styles applied.

![image](https://user-images.githubusercontent.com/5596960/106616124-78f38d00-656d-11eb-8242-604f87157ef0.png)


drawbacks:
- mounting takes a while, and for a moment cms element is not visible at all, can be applied only for the "elements" or the smaller cms components

https://drive.google.com/file/d/1y2IzdRqsYCuubOyPK8ZmkaXhNlAsM_hW/view

**Comment by niklaswolf** [comment]
@mkucmus I think there the bug is in the `renderHtml` function which comes from a third party package.

You can reproduce this with the CmsElementText component as follows: 
This works! SSR Html contains the inline styles.
```
render(h, context) {
    const config = {...}
    // const rawHtml = context.props.content?.data?.content
    
    const element = h("h1",{style: "text-align: center; color: #fff", domProps: {
        innerHTML: 'HElloooooo'
      }});
    return element;
  },
  ```
This doesn't work, inline styles are not in the HTML
```
render(h, context) {
    const config = {...}
    // const rawHtml = context.props.content?.data?.content

    const rendered = renderHtml("<h1 style='text-align: center; color: #fff'>HElloooooo</h1>", config, h, context)
    return rendered;
  },
  ```
I don't see any big difference apart from the `renderHtml` function, so it's likely that its the culprit!

**Comment by mkucmus** [comment]
Yeah, you are right, the `renderHtml` is behaving odd, but in the end it uses a `h` method under the hood, which is an alias for `createElement` built-in vue method. But the thing that the renderHtml does is more complex, does much much more. it deserializes the raw text into vue components, using additional mapping and the issue itself is strictly related with component life-cycle in that specific case (maybe we have found an edge case?) - in my opinion ;) 

Although the library is cool enough to leave it here for custom mapping, it's not possible to fix it in a project's scope. But we still can create an issue about the problem we have on SSR. 
Still as a workaround we could use a mounted hack to get rid of it till the problem is solved on library side.

@niklaswolf what about creating an issue in the author's repository and set is as a blocker for that issue? Or we can just wait for good people to write the similar library without any errors for us :D

**Comment by niklaswolf** [comment]
@mkucmus yeah we should create an issue and see if we get a response. What's really strange is that this is somehow related to SSR and component lifecycle as you said, but having a quick scan of the code, I don't see any vue lifecycle specific code. But yeah, let's see what he says. I'll create an issue and link this one here.

**Comment by Rigo-m** [comment]
Thanks for the heads up guys, I'll fix the issue this morning or this evening at most :)

**Comment by Rigo-m** [comment]
Fixed with html-to-vue@1.3.0
